pkg/controller/account/iam.go

```diff
@@ -731,3 +731,199 @@ func (r *ReconcileAccount) createManagedOpenShiftSupportRole(reqLogger logr.Logg
+
+	var iamAccessKeyOutput *iam.CreateAccessKeyOutput
+
+	if !iamUserExists {
+		// If user doesn't exist, create new IAM user
+		CreateUserOutput, err := awsclient.CreateIAMUser(reqLogger, awsClient, account, iamUserName, managedTags, customTags)
+		// Err is handled within the function and returns a error message
+		if err != nil {
+			return err
+		}
+
+		// Extract iam.User as pointer
+		newIAMUser := CreateUserOutput.User
+
+		reqLogger.Info(fmt.Sprintf("Attaching Admin Policy to IAM user %s", aws.StringValue(newIAMUser.UserName)))
+
+		// Setting IAM user policy
+		_, err = AttachAdminUserPolicy(awsClient, newIAMUser)
+		if err != nil {
+			errMsg := fmt.Sprintf("Failed to attach admin policy to IAM user %s", aws.StringValue(newIAMUser.UserName))
+			reqLogger.Error(err, errMsg)
+			return err
+		}
+
+		// Create new access key
+		iamAccessKeyOutput, err = CreateUserAccessKey(awsClient, newIAMUser)
+		if err != nil {
+			reqLogger.Error(err, "failed to create IAM access key", "IAMUser", newIAMUser.UserName)
+			return err
+		}
+	} else {
+		// If user exists extract iam.User pointer and rotate access key
+		currentIAMUser := iamUserExistsOutput.User
+		iamAccessKeyOutput, err = r.RotateIAMAccessKeys(reqLogger, awsClient, account, currentIAMUser)
+		if err != nil {
+			errMsg := fmt.Sprintf("Unable to rotate access keys for IAM user: %s", aws.StringValue(currentIAMUser.UserName))
+			reqLogger.Error(err, errMsg)
+			return err
+		}
+	}
+	// Update access keys on kubernetes secrets
+	err = r.updateIAMUserSecret(reqLogger, account, kubeSecretNamespacedName, iamAccessKeyOutput)
+	if err != nil {
+		errMsg := fmt.Sprintf("Unable to update kubernetes secret: %s", kubeSecretNamespacedName.Name)
+		reqLogger.Error(err, errMsg)
+		return err
+	}
+
+	// Return secret name
+	return nil
+}
+
+func (r *ReconcileAccount) updateIAMUserSecret(reqLogger logr.Logger, account *awsv1alpha1.Account, secretName types.NamespacedName, createAccessKeyOutput *iam.CreateAccessKeyOutput) error {
+
+	// Fill in the secret data
+	userSecretData := map[string][]byte{
+		"aws_user_name":         []byte(*createAccessKeyOutput.AccessKey.UserName),
+		"aws_access_key_id":     []byte(*createAccessKeyOutput.AccessKey.AccessKeyId),
+		"aws_secret_access_key": []byte(*createAccessKeyOutput.AccessKey.SecretAccessKey),
+	}
+
+	// Create new secret
+	iamUserSecret := CreateSecret(secretName.Name, secretName.Namespace, userSecretData)
+
+	// Set controller as owner of secret
+	if err := controllerutil.SetControllerReference(account, iamUserSecret, r.scheme); err != nil {
+		return err
+	}
+
+	// Update secret
+	updateErr := r.Client.Update(context.TODO(), iamUserSecret)
+	if updateErr != nil {
+		failedToUpdateUserSecretMsg := fmt.Sprintf("Failed to create secret %s", iamUserSecret.Name)
+		utils.SetAccountStatus(account, failedToUpdateUserSecretMsg, awsv1alpha1.AccountFailed, "Failed")
+		err := r.Client.Status().Update(context.TODO(), account)
+		if err != nil {
+			return err
```

Comment:
wouldn't we still be interested in passing up the updateErr, or an error that contains both error messages?